extensions/wikia/ArticleAsJson/ArticleAsJson.class.php

```diff
@@ -766,7 +766,7 @@ private static function getMediaDetailWithSizeFallback(
 				\Wikia\Logger\WikiaLogger::instance()->notice(
 					'ArticleAsJson - Media height was empty - fallback to fallbackSize',
 					[
-						'media_details' => $mediaDetail,
+						'media_details' => json_encode( $mediaDetail ),
```

Comment:
Can we make this sort of preprocessing automatic? In the logging method, we could iterate over the context and encode all array keys.

Reply:
technically it is possible. but still there are low-cardinality array-type fields which I use for filtering, like `@context.edit_status. is_ok `, `@context.ex.message` `@context.reqStatus.is_ok` to name a few.
Now, we could have a rule that we encode arrays having more than 4 elements. But now we get into a situation that if for the same field we have arrays of different sizes, then sometimes you encode it as array and sometimes as strings. And due to type conflicts ES will drop the incompatible values.
So I think patching the high-cardinality arrays at the source is actually easier.

Reply:
Makes sense, thanks!